lib/Doctrine/DBAL/Portability/Statement.php

```diff
@@ -213,6 +213,55 @@ protected function fixRow($row, $iterateRow, $fixCase)
     }
 
     /**
+     * Fixes column values according to portability settings
+     *
+     * @param mixed[] $values Column values indexed by row number
+     * @param bool $emptyToNull Should empty strings be converted to NULL
+     * @param bool $rTrim Should strings be right-trimmed
+     * @return mixed[]
+     */
+    private function fixColumn(array $values, $emptyToNull, $rTrim)
+    {
+        if (!$emptyToNull && !$rTrim) {
```

Comment:
I think ideally (if [performance allows](http://stackoverflow.com/questions/18144782/performance-of-foreach-array-map-with-lambda-and-array-map-with-static-function/26527704#26527704)), we could have just 2-3 lower level functions which convert keys and values, and combine them together in reusable combinations based on `fetch*(PDO::FETCH_*)`. But let's try your suggestion first.